internal/integration/collection_test.go

```diff
@@ -2028,6 +2030,386 @@ func TestCollection(t *testing.T) {
 	})
 }
 
+func TestBypassEmptyTsReplacement(t *testing.T) {
+	mt := mtest.New(t, mtest.NewOptions().CreateClient(false).MinServerVersion("5.0"))
+
+	marshalValue := func(val interface{}) bson.RawValue {
+		t.Helper()
+
+		valType, data, err := bson.MarshalValue(val)
+		require.NoError(t, err, "MarshalValue error: %v", err)
+		return bson.RawValue{
+			Type:  valType,
+			Value: data,
+		}
+	}
+
+	mt.Run("insert one", func(mt *mtest.T) {
+		doc := bson.D{{"x", 42}}
+
+		newOpts := func(option bson.D) *options.InsertOneOptionsBuilder {
+			opts := options.InsertOne()
+			err := xoptions.SetInternalInsertOneOptions(opts, "addCommandFields", option)
+			require.NoError(mt, err, "unexpected error: %v", err)
+			return opts
+		}
+
+		testCases := []struct {
+			name     string
+			opts     *options.InsertOneOptionsBuilder
+			expected bson.RawValue
+		}{
+			{
+				"empty",
+				nil,
+				bson.RawValue{},
+			},
```

Comment:
Optional: Keyed struct literals are easier to use. Consider including the key names.